### PR TITLE
compressor: use DEV/APP instead of SRC/DST for addresses and ports

### PR DESCRIPTION
**Pull request by miri64**
This adopts libSCHC to use the address semantics lined out in RFC 8724, section 10.7 and 10.9.

Other than before, just swapping the address fields before compression and after decompression does not work, as then the address would be the wrong way around in the compression residue. As such, we need to abort iterative compression/decompression in the DOWN case, but save the address field rule index, swap the rule iteration for those instead, and then continue iterative compression, in case there are still remaining fields (e.g. as in the UDP header).

I also added some bug fixes of bugs in the compression I noticed during testing. I can put those in a separate PR, if the actual fix needs some more discussion.

### CODE REVIEW

**Comment by bartmoons** [comment]
The compression/decompression example only works in the upward direction. Could you check the downward direction? I get the following with the example rules that you included:
```
schc_compress(): 
schc_find_rule_from_header(): skipped rule 01, IPv6 Application Prefix does not match
schc_find_rule_from_header(): skipped rule 02, IPv6 Application Prefix does not match
schc_find_rule_from_header(): skipped rule 03, IPv6 Field Length does not match
schc_find_rule_from_header(): skipped rule 04, IPv6 Device Prefix does not match
schc_find_rule_from_header(): skipped rule 01, UDP Application Port does not match
schc_find_rule_from_header(): skipped rule 02, UDP Application Port does not match
schc_find_rule_from_header(): skipped rule 03, UDP Application Port does not match
schc_find_rule_from_header(): skipped rule 04, UDP Application Port does not match
schc_compress(): CoAP rule ptr=0x55a659081140 
schc_compress(): no rule was found 
```
I also noticed that the decompression of the resulting uncompressed packet is wrong, but I just provided a fix for that on the master branch.

**Comment by miri64** [comment]
That upwards works, does not surprise me: it basically works still in the same way as it works in current master.

Do you have a specific packet and rule set I can test for that? I have tested both [downward compression](https://github.com/RIOT-OS/RIOT/blob/264fc92d4ff22aca01c93787692cc43f9c276099/tests/pkg_libschc/tests/01-run.py#L724-L745) and [decompression](https://github.com/RIOT-OS/RIOT/blob/264fc92d4ff22aca01c93787692cc43f9c276099/tests/pkg_libschc/tests/01-run.py#L748-L767) with [this rule](https://github.com/RIOT-OS/RIOT/blob/264fc92d4ff22aca01c93787692cc43f9c276099/pkg/libschc/include/rules/rules.h#L234-L246) and it worked.

**Comment by bartmoons** [comment]
Yes, the ones that are provided in `compress.c` and `rules_example.h` from the examples.

**Comment by miri64** [comment]
Ah, I think the rules might need some adoption in that case. I changed the names, but not the semantics

**Comment by miri64** [comment]
I fixed the examples and rules to fit how the rules now work. I could not compile `interop.c` and there was a compile warning for `lwm2m.c`, so I did not test those.

**Comment by bartmoons** [comment]
I am ignoring those two examples for now. Thanks for the fix, the rules makes sense to me now.

**Comment by miri64** [comment]
Thanks you for the review and testing!

File: rules/rule_config_example.h [hunk 1]
```diff
@@ -9,4 +9,4 @@
  *
  */
 
-#include "rules.h"
+#include "rules_example.h"
```
Comment (miri64) [inline rules/rule_config_example.h:12]: Oops, I did not intent to add this... Will revert in #32.

**Pull request merged**